src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -80,22 +80,25 @@ private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
 
             if (node.IsParentKind(SyntaxKind.ReturnStatement))
             {
-                var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsReturnableConstruct());
+                var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsKind(
+                    SyntaxKind.MethodDeclaration, SyntaxKind.PropertyDeclaration, SyntaxKind.ParenthesizedLambdaExpression, SyntaxKind.SimpleLambdaExpression,
+                    SyntaxKind.LocalFunctionStatement, SyntaxKind.AnonymousMethodExpression, SyntaxKind.ConstructorDeclaration, SyntaxKind.DestructorDeclaration,
+                    SyntaxKind.OperatorDeclaration, SyntaxKind.IndexerDeclaration, SyntaxKind.EventDeclaration));
 
                 if (containingMember == null)
```

Comment:
nit: this is now unnecessary with pattern matching

Reply:
I don't want to use pattern matching here because it involves many type tests, repeated for each level of ancestry.

Reply:
I meant the pattern matching that you're already using below. Not that you should use it above as well.

Reply:
> it involves many type tests, repeated for each level of ancestry

Is there a performance problem with that? I've seen `GetAncestorOrSelf<T>` and such used all over the place.

Reply:
Oh, right. That one (below) is fine :-)

Reply:
Yes, I suspect there would be some performance impact. My prior could be changed with some micro-benchmark, once someone takes the time to check.

The problem is not with `GetAncestor` method itself. But since it's a loop, and you want to make iterations cheaper when in such a tight loop.